Code/Tools/AssetProcessor/Platform/Linux/native/FileWatcher/FileWatcher_linux.cpp

```diff
@@ -15,155 +18,133 @@
 #include <sys/inotify.h>
 
 
-static constexpr int s_handleToFolderMapLockTimeout = 1000;      // 1 sec timeout for obtaining the handle to folder map lock
-static constexpr size_t s_iNotifyMaxEntries = 1024 * 16;         // Control the maximum number of entries (from inotify) that can be read at one time
-static constexpr size_t s_iNotifyEventSize = sizeof(struct inotify_event);
-static constexpr size_t s_iNotifyReadBufferSize = s_iNotifyMaxEntries * s_iNotifyEventSize;
+static constexpr size_t s_inotifyMaxEntries = 1024 * 16;         // Control the maximum number of entries (from inotify) that can be read at one time
+static constexpr size_t s_inotifyEventSize = sizeof(struct inotify_event);
+static constexpr size_t s_inotifyReadBufferSize = s_inotifyMaxEntries * s_inotifyEventSize;
 
-struct FolderRootWatch::PlatformImplementation
+bool FileWatcher::PlatformImplementation::Initialize()
 {
-    PlatformImplementation() = default;
-
-    int                         m_iNotifyHandle = -1;
-    QMutex                      m_handleToFolderMapLock;
-    QHash<int, QString>         m_handleToFolderMap;
+    if (m_inotifyHandle < 0)
+    {
+        // The CLOEXEC flag prevents the inotify watchers from copying on fork/exec
+        m_inotifyHandle = inotify_init1(IN_CLOEXEC);
+
+#if defined(AZ_ENABLE_TRACING)
+        const auto err = errno;
+        AZStd::fixed_string<255> errorString;
+#endif
+        AZ_Warning("FileWatcher", (m_inotifyHandle >= 0), "Unable to initialize inotify, file monitoring will not be available: %s\n", strerror_r(err, errorString.data(), errorString.capacity()));
```

Comment:
`AZ_Warning` is only enabled when the `AZ_ENABLE_TRACING` define is set, which means this is just saving an assignment to an `err` integer variable, for the user of the macro

Personally I don't see the need for the `#if defined(AZ_ENABLE_TRACING)`

Reply:
I think ```[[maybe_unused]] AZStd::fixed_string<255> errorString;``` is fine here so we don't need more #ifdef guards
```suggestion

        [[maybe_unused]] AZStd::fixed_string<255> errorString;
        AZ_Warning("FileWatcher", (m_inotifyHandle >= 0), "Unable to initialize inotify, file monitoring will not be available: %s\n", strerror_r(errno, errorString.data(), errorString.capacity()));
```